File: gachisa-backend/src/main/java/com/gachisa/order/dto/OrderListResponse.java
package com.gachisa.order.dto;

import com.gachisa.order.entity.Order;
import java.util.List;
import org.springframework.data.domain.Page;

public record OrderListResponse(
        List<OrderResponse> content,
        int page,
        int size,
        long totalElements,
        int totalPages
) {

    public static OrderListResponse from(Page<Order> orders) {
        return new OrderListResponse(
                orders.getContent().stream().map(OrderResponse::from).toList(),
                orders.getNumber(),
                orders.getSize(),
                orders.getTotalElements(),
                orders.getTotalPages()
        );
    }
}

File: gachisa-backend/src/main/kotlin/com/gachisa/order/dto/OrderListResponse.kt
package com.gachisa.order.dto

import com.gachisa.order.entity.Order
import org.springframework.data.domain.Page

data class OrderListResponse(
    val content: List<OrderResponse>,
    val page: Int,
    val size: Int,
    val totalElements: Long,
    val totalPages: Int,
) {
    companion object {
        fun from(orders: Page<Order>): OrderListResponse = OrderListResponse(
            content = orders.content.map(OrderResponse::from),
            page = orders.number,
            size = orders.size,
            totalElements = orders.totalElements,
            totalPages = orders.totalPages,
        )
    }
}
